### PR TITLE
add checkout commit option

### PR DESCRIPTION
**Pull request by shakednai1**

### CODE REVIEW

File: engine/src/Branch.java [hunk 1]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
...
+                    deleteSubFilesRec(item);
+                }
+            }
+            else{
+                Utils.deleteFile(item.getPath());
+            }
+        }
+    }
+
+    private void deleteSubFilesRec(File folder) {
```
Comment (AlonaTy) [inline engine/src/Branch.java]: should delete all current repo folder?
if so, use  
```suggestion
    FileUtils.deleteDirectory(destTestsDir);
```
do it all for you in one shot

File: engine/src/Branch.java [hunk 2]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
...
+        recursiveOpenAllFiles(getRootSha1(), Settings.repositoryFullPath);
+    }
+
+    private void clearCurrentWC() {
+        File directory = new File(Settings.repositoryFullPath);
+        File[] listOfItems = directory.listFiles();
+        for(File item: listOfItems){
+            if(item.isDirectory()){
+                if(!item.getName().equals(Settings.gitFolder)){
+                    deleteSubFilesRec(item);
```
Comment (AlonaTy) [inline engine/src/Branch.java:63]:
```suggestion
                    File destDir = new File("C://repo_path");
                    FileUtils.deleteDirectory(destDir);
```

do it for you

File: engine/src/Branch.java [hunk 3]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
...
+            }
+            else{
+                Utils.deleteFile(item.getPath());
+            }
+        }
+        Utils.deleteFile(folder.getPath());
+    }
+
+
+    private String getRootSha1(){
```
Comment (AlonaTy) [inline engine/src/Branch.java:87]: I think this function should be a static function at `Commit` class that gets the commit_sha1_path.
Those files are writen by the Commit class and which knows ythe logic of those files. 

What do you think?

File: engine/src/Branch.java [hunk 4]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
...
+        //String newCommitDir = Settings.objectsFolderPath + head.commitSha1 + "tmp" + ".txt";
+        Utils.unzip(Settings.objectsFolderPath + head.commitSha1 + ".zip", Settings.objectsFolderPath);
```
Comment (AlonaTy) [inline engine/src/Branch.java:89]: I prefer it to be in a different older (maybe a dedicated folder under `.objects`) that saves we extract the shit to

Comment (AlonaTy) [inline engine/src/Branch.java:89]: Format of the sha file should go to a setting function so we will use that format everywhere

File: engine/src/Branch.java [hunk 5]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
...
+            return contents.split(Settings.delimiter)[0];
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private void recursiveOpenAllFiles(String sha1, String location){
+        //String newFile = Settings.objectsFolderPath + sha1 + "tmp" + ".txt";
+        Utils.unzip(Settings.objectsFolderPath + sha1 + ".zip", Settings.objectsFolderPath);
```
Comment (AlonaTy) [inline engine/src/Branch.java:103]: dedicated folder will be better

File: engine/src/Branch.java [hunk 6]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
...
+                }
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void openFile(String itemSha1, String filePath){
```
Comment (AlonaTy) [inline engine/src/Branch.java:129]: function's code is pretty stright forward , a dedicated function is redundant

File: engine/src/Branch.java [hunk 7]
```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
     public Commit getStartCommit(){
         return startCommit;
     }
+
+    public void open() {
```
Comment (AlonaTy) [inline engine/src/Branch.java:52]: Our `BranchManager` knows to `commit` the current branch and I think that it also have to know how to `loadFromCommit(String commitSHA1)`, and also all the objects that commit themselves need to know how deploy themselves and create coresponding objects from that deploy.

Lets talk about that

Comment (shakednai1) [reply engine/src/Branch.java:52]: the checkout action is between branches as the user experience it (behind the scene of course its switch between commits)
I think we should stay this method as branch method

File: engine/src/Branch.java (hunk 5 above)
Comment (shakednai1) [reply engine/src/Branch.java:103]: why? its add more recursive deletion for the new folder

File: engine/src/Branch.java (hunk 4 above)
Comment (shakednai1) [reply engine/src/Branch.java:89]: why? its add more recursive deletion for the new folder

File: engine/src/Branch.java (hunk 3 above)
Comment (shakednai1) [reply engine/src/Branch.java:87]: same as above:
the checkout action is between branches as the user experience it (behind the scene of course its switch between commits)

**Pull request merged**